cpp/src/cylon/table.cpp

```diff
@@ -583,7 +548,13 @@ Status DetermineSplitPoints(
   int num_split_points =
       std::min(merged_table->Rows(), (int64_t)ctx->GetWorldSize() - 1);
 
-  return SampleTableUniform(merged_table, num_split_points, sort_columns, split_points, ctx);
+
+
+  auto status = util::SampleTableUniform(merged_table->get_table(), num_split_points, sort_columns,
+                                         const_cast<std::shared_ptr<arrow::Table> &>(split_points->get_table()), ctx);
+
+  return cylon::Status(static_cast<int>(status.code()), status.message());
```

Comment:
@mstaylor this could be problematic. Hmm... This will actually create a NPE AFAIU. 
```suggestion
  std::shared_ptr<arrow::Table> atable;
  RETURN_CYLON_STATUS_IF_ARROW_FAILS(util::SampleTableUniform(merged_table->get_table(),
  num_split_points, sort_columns, atable, ctx);
  
  return Table::FromArrowTable(ctx, std::move(atable), split_points);
```

Reply:
Completed